crates/hir_ty/src/infer/coerce.rs

```diff
@@ -41,6 +63,7 @@ impl<'a> InferenceContext<'a> {
     ///  - if we were concerned with lifetime subtyping, we'd need to look for a
     ///    least upper bound.
     pub(super) fn coerce_merge_branch(&mut self, id: Option<ExprId>, ty1: &Ty, ty2: &Ty) -> Ty {
+        // TODO
```

Comment:
I am not sure how to do this one here, since we are coercing two differen types to one we cant return a simpel CoercionResult here since we get an adjustments per type each right?

Reply:
Yeah, that's what `CoerceMany` is for in rustc. In a match, you would need to add the new adjustments to each of the previous branches.

Reply:
Ah I see, so we wanna adapt that one as well I assume